src/Toast.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Fade from './Fade';
+import Header from './ToastHeader';
+import Body from './ToastBody';
+import { createBootstrapComponent } from './ThemeProvider';
+import ToastDialog from './ToastDialog';
+
+const propTypes = {
+  /**
+   * @default 'toast'
+   */
+  bsPrefix: PropTypes.string,
+
+  /**
+   * A Component type that provides the toast content Markup. This is a useful
+   * prop when you want to use your own styles and markup to create a custom
+   * toast component.
+   */
+  dialogAs: PropTypes.elementType,
+
+  /**
+   * Apply a CSS fade transition to the toast
+   */
+  animation: PropTypes.bool,
+
+  /**
+   * Auto hide the toast
+   */
+  autohide: PropTypes.bool,
+
+  /**
+   * Delay hiding the toast (ms)
+   */
+  delay: PropTypes.number,
+
+  /**
+   * A Callback fired when the close button is clicked.
+   */
+  onClose: PropTypes.func,
+
+  /**
+   * When `true` The modal will show itself.
+   */
+  show: PropTypes.bool,
+
+  /** A `react-transition-group` Transition component used to animate the Toast on dismissal. */
+  transition: PropTypes.elementType,
+};
+
+const defaultProps = {
+  animation: true,
+  autohide: false,
+  delay: 3000,
+  dialogAs: ToastDialog,
+  show: true,
+  transition: Fade,
+};
+
+const Toast = ({ children, dialogAs: Dialog, ...props }) => (
+  <Dialog {...props}>{children}</Dialog>
```

Comment:
since there is no logic in this component it seems like the `dialogAs` is probably an unnecessarily customization...what about removing `ToastDialog` and moving that code here as just `Toast`?